typescript/core/src/v0/ChunkCursor.ts

```diff
@@ -87,12 +103,22 @@ export class ChunkCursor {
     }
     const logTimeA = cursorA.records[cursorA.index]![0];
     const logTimeB = cursorB.records[cursorB.index]![0];
-    if (logTimeA !== logTimeB) {
-      return Number(logTimeA - logTimeB);
-    }
 
-    // Break ties by chunk offset in the file
-    return Number(this.chunkIndex.chunkStartOffset - other.chunkIndex.chunkStartOffset);
+    if (reverse) {
+      if (logTimeA !== logTimeB) {
+        return Number(logTimeB - logTimeA);
+      }
+
+      // Break ties by chunk offset in the file
+      return Number(other.chunkIndex.chunkStartOffset - this.chunkIndex.chunkStartOffset);
+    } else {
+      if (logTimeA !== logTimeB) {
+        return Number(logTimeA - logTimeB);
+      }
+
+      // Break ties by chunk offset in the file
+      return Number(this.chunkIndex.chunkStartOffset - other.chunkIndex.chunkStartOffset);
+    }
```

Comment:
Not quite. There's this bit of code that does not change under the reverse flag:

```
  // If a cursor has not loaded message indexes, sort it first so it can get loaded and re-sorted.
  // If it overlaps with other cursors, we won't know its order relative to other cursors until we
  // have the message indexes.
 if (!this.messageIndexCursors) {
      return -1;
    }
    if (!other.messageIndexCursors) {
      return 1;
    }
```